src/main/java/org/apache/sysml/hops/codegen/opt/PlanSelectionFuseCostBasedV2.java

```diff
@@ -215,20 +215,23 @@ private void selectPlans(CPlanMemoTable memo, PlanPartition part)
 	 * @return optimal assignment of materialization points
 	 */
 	private static boolean[] enumPlans(CPlanMemoTable memo, PlanPartition part, StaticCosts costs, 
-		ReachabilityGraph rgraph, InterestingPoint[] matPoints, int off, double bestC) 
+		ReachabilityGraph rgraph, InterestingPoint[] matPoints, int off, double bestC)
 	{
 		//scan linearized search space, w/ skips for branch and bound pruning
 		//and structural pruning (where we solve conditionally independent problems)
 		//bestC is monotonically non-increasing and serves as the upper bound
-		long len = (long)Math.pow(2, matPoints.length-off);
+		long len = 1L << matPoints.length-off;
```

Comment:
Thanks for the experiment. Well, I guess this cannot be done automatically here because the two expressions give different results at the high end (which is irrelevant for us). However, note that evaluating 50M plans would take you about 1000s, which means that 3s simply don't matter.